apps/andi/lib/andi_web/input_schemas/ingestion_metatadata_form_schema.ex

```diff
@@ -26,55 +26,36 @@ defmodule AndiWeb.InputSchemas.IngestionMetadataFormSchema do
     :targetDataset
   ]
 
-  def changeset(changes), do: changeset(%__MODULE__{}, changes)
-
-  def changeset(%__MODULE__{} = current, changes) do
-    current
-    |> cast(changes, @cast_fields, empty_values: [])
-    |> validate_required(@required_fields, message: "is required")
-    |> validate_top_level_selector()
-    |> target_dataset_exists()
-  end
-
-  def changeset_from_andi_ingestion(ingestion) do
-    ingestion = StructTools.to_map(ingestion)
-
-    changeset(ingestion)
-  end
-
-  defp target_dataset_exists(changeset) do
-    validate_change(changeset, :targetDataset, fn :targetDataset, targetDataset ->
-      case Andi.InputSchemas.Datasets.get(targetDataset) do
-        nil ->
-          [
-            targetDataset: "Dataset with id: #{targetDataset} does not exist. It may have been deleted."
-          ]
-
-        _ ->
-          []
-      end
-    end)
-  end
-
-  @spec changeset_from_form_data(any) :: Ecto.Changeset.t()
-  def changeset_from_form_data(form_data) do
-    form_data
-    |> AtomicMap.convert(safe: false, underscore: false)
-    |> changeset()
+  def extract_from_ingestion_changeset(%Ecto.Changeset{data: %Andi.InputSchemas.Ingestion{}} = ingestion_changeset) do
```

Comment:
Now that the parent changeset (Ingestion) is the source of truth, we need to pass down the data and errors appropriately.